source/payload-interfaces/index.rst

```diff
@@ -360,6 +360,158 @@ Set to 0 to use the default baud rate 115200.
+    //
+    UINT64 Translation;
+  } PLD_PCI_ROOT_BRIDGE_APERTURE;
+  #pragma pack()
+
+**Member Description**
+
+``PldHeader``
+
+PldHeader.Revision is 1.
+
+PldHeader.Length is 5 + ``Count`` * sizeof (PLD_PCI_ROOT_BRIDGE).
+
+``Count``
+
+Count of root bridges. Number of elements in ``RootBridge`` array.
+
+``RootBridge[i].Segment``
+
+Segment number of the root bridge.
+
+``RootBridge[i].Supports``
+
+Supported attributes. Refer to EFI_PCI_ATTRIBUTE_xxx used by GetAttributes() and SetAttributes() in EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL
+defined in PI Specification.
+
+``RootBridge[i].Attributes``
+
+Initial attributes. Refer to EFI_PCI_ATTRIBUTE_xxx used by GetAttributes() and SetAttributes() in EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL
+defined in PI Specification.
+
+``RootBridge[i].DmaAbove4G``
+
+Root bridge supports DMA above 4GB memory when it's TRUE.
+
+``RootBridge[i].NoExtendedConfigSpace``
+
+Root bridge supports 256-byte configuration space only when it's TRUE.
+Root bridge supports 4K-byte configuration space when it's FALSE.
+
+``RootBridge[i].ResourceAssigned``
+
+Bus/IO/MMIO resources for root bridge have been assigned when it's TRUE.
+
+``RootBridge[i].AllocationAttributes``
+
+Allocation attributes. Refer to EFI_PCI_HOST_BRIDGE_COMBINE_MEM_PMEM and EFI_PCI_HOST_BRIDGE_MEM64_DECODE used by GetAllocAttributes()
+in EFI_PCI_HOST_BRIDGE_RESOURCE_ALLOCATION_PROTOCOL defined in PI Specification.
+
+``RootBridge[i].Bus``
+
+Bus aperture which can be used by the root bridge when ``RootBridge[i].ResourceAssigned`` is FALSE.
+Bus aperture which has been used by the root bridge when ``RootBridge[i].ResourceAssigned`` is TRUE.
+
+``RootBridge[i].Io``
+
+IO aperture which can be used by the root bridge when ``RootBridge[i].ResourceAssigned`` is FALSE.
+IO aperture which has been used by the root bridge when ``RootBridge[i].ResourceAssigned`` is TRUE.
+
+``RootBridge[i].Mem``
+
+MMIO aperture below 4GB which can be used by the root bridge when ``RootBridge[i].ResourceAssigned`` is FALSE.
+MMIO aperture below 4GB which has been used by the root bridge when ``RootBridge[i].ResourceAssigned`` is TRUE.
+
+``RootBridge[i].MemAbove4G``
+
+MMIO aperture above 4GB which can be used by the root bridge when ``RootBridge[i].ResourceAssigned`` is FALSE.
+MMIO aperture above 4GB which has been used by the root bridge when ``RootBridge[i].ResourceAssigned`` is TRUE.
+
+``RootBridge[i].PMem``
+
+Prefetchable MMIO aperture below 4GB which can be used by the root bridge when ``RootBridge[i].ResourceAssigned`` is FALSE.
+Prefetchable MMIO aperture below 4GB which has been used by the root bridge when ``RootBridge[i].ResourceAssigned`` is TRUE.
+
+``RootBridge[i].PMemAbove4G``
+
+Prefetchable MMIO aperture above 4GB which can be used by the root bridge when ``RootBridge[i].ResourceAssigned`` is FALSE.
+Prefetchable MMIO aperture above 4GB which has been used by the root bridge when ``RootBridge[i].ResourceAssigned`` is TRUE.
+
+``RootBridge[i].HID``

```

Comment:
Does payload do anything here with ACPI table ？
If the ID does not match,  what would happen ?

Reply:
HID is for payload to construct the device path of Pci Root bridge.
It doesn't connect to ACPI table.

It's a good question, so maybe we can drop this field. use EISA_PNP_ID(0x0A08) or EISA_PNP_ID(0x0A03) depending on NoExtendedConfigSpace field.

But, I can only find 0A03 in internet (describing PCI BUS) and cannot find 0A08. But today's firmware code uses 0A08 for PCIE BUS.

Reply:
http://download.microsoft.com/download/1/6/1/161ba512-40e2-4cc9-843a-923143f3456c/devids.txt only lists 0A03 but no 0A08. Anyway, I will remove the HID and UID.

Reply:
From OS point of view, firmware is responsible for providing consistent info regarding HID and UID.
I also checked today's server code, it constructs UID using the stack ID. The value for UID used in ASL and PciHostBridgeDxe driver is the same.
so, I think it'd better keep the HID/UID consistency otherwise we don't know whether OS might run wrong.

Reply:
OK.  I got it.
Just wondering how payload will use this HID/UID info if it does not modify the DSDT table at all.  EX,  you mentioned server will constructs UID using stack ID, where is the UID used ?

Reply:
Maurice,
I am not quite clear about your question.

Bootloader constructs the ACPI table that describes the PCI BUSes. The HID/UID should be specified in the ACPI table.

The HID/UID in this interface guides UEFI payload to construct the device path of each root bridge. I want bootloader pass the information to payload to make sure the device path constructed by payload matches to the ACPI table.

Because UEFI OS loader can consume the ACPI table and the device path before ExitBootService, I want to provide a consistent HID/UID from ACPI table and device path.

Reply:
OK,  I think that is fine.